instrumentation/grape/lib/opentelemetry/instrumentation/grape/handler.rb

```diff
@@ -28,15 +26,68 @@ def subscribe
 
           private
 
-          def endpoint_start_process(name, start, finish, id, payload); end
+          def endpoint_run(name, start, finish, id, payload)
```

Comment:
so this has to be the parent instrumentation point. the problem is that the callback is executed after the endpoint has finished running. Thus we need to have a custom subscriber that will create the root span when the instrumentation begins. and finish it when the instrumentation ends. All other callbacks will create spans that are children of this root span.

I am sure you had thought of this, but just wanted to lay my thoughts out there.